projects/libs/core-library-typescript/src/common-library/startup-check-runner.ts

```diff
@@ -0,0 +1,52 @@
+import { StartupCheck } from "./startup-check";
+import { StartupStatus, STARTUP_STATUS } from "./startup-status";
+import { logger } from "../common-library";
+
+export class StartupCheckRunner {
+  name: string;
+  startupCheckers: StartupCheck[] = [];
+
+  constructor(name: string, startupCheckers: StartupCheck[]) {
+    this.name = name;
+    this.startupCheckers = startupCheckers;
+  }
+
+  run(): Promise<StartupStatus> {
+
+    const startupCheckPromises = this.startupCheckers.map((s: StartupCheck) => s.startupCheck());
+    return Promise.all(startupCheckPromises)
+      .then(this.mergeStartupStatuses)
+      .catch(err => {
+        logger.error(err);
+        return <StartupStatus>{ status: STARTUP_STATUS.FAIL, message: err.message };
+      });
+  }
+
+  private mergeStartupStatuses = (startupStatuses: StartupStatus[]): StartupStatus => {
```

Comment:
Extract as a method.

Reply:
What is the problem with this impl?

Reply:
It's a function defined inside a class and not a method, which is very unusual and does not serve any practical purpose.